src/AttachVS/AttachVs.cs

```diff
@@ -0,0 +1,309 @@
+
+                                while (lpn.MoveNext())
+                                {
+                                    var pn = Convert.ToInt32(lpn.Current.GetType().InvokeMember("ProcessID", BindingFlags.GetProperty, null, lpn.Current, null));
+
+                                    if (pn == pid)
+                                    {
+                                        lpn.Current.GetType().InvokeMember("Attach", BindingFlags.InvokeMethod, null, lpn.Current, null);
+                                        return true;
+                                    }
+                                }
+                            }
+                            catch (COMException ex)
+                            {
+                                Trace($"ComException: Tetrying in 250ms.\n{ex}");
+                                Thread.Sleep(250);
+                            }
+                        }
+                        Marshal.ReleaseComObject(moniker[0]);
+
+                        break;
+                    }
+
+                    Marshal.ReleaseComObject(moniker[0]);
+                }
+                return false;
+            }
+            finally
+            {
+                if (enumMoniker != null)
+                {
+                    try
+                    {
+                        Marshal.ReleaseComObject(enumMoniker);
+                    }
+                    catch { }
+                }
+                if (runninObjectTable != null)
+                {
+                    try
+                    {
+                        Marshal.ReleaseComObject(runninObjectTable);
+                    }
+                    catch { }
+                }
+                if (bindCtx != null)
+                {
+                    try
+                    {
+                        Marshal.ReleaseComObject(bindCtx);
+                    }
+                    catch { }
+                }
+            }
+        }
+
+        private static Process GetVsFromPid(Process process)
+        {
+            var parent = process;
+            while (!IsVsOrNull(parent))
+            {
+                parent = GetParentProcess(parent);
+            }
+
+            return parent;
+        }
+
+        private static bool IsVsOrNull(Process process)
+        {
+            if (process == null)
+            {
+                Trace("Parent process is null..");
+                return true;
+            }
+
+            try
+            {
+                var isVs = process.ProcessName.Equals("devenv", StringComparison.InvariantCultureIgnoreCase);
+                if (isVs)
+                {
+                    Trace($"Process {process.ProcessName} ({process.Id}) is VS.");
+                }
+                else
+                {
+                    Trace($"Process {process.ProcessName} ({process.Id}) is not VS.");
+                }
+
+                return isVs;
+            }
+            catch
```

Comment:
Why would the above `try` block throw anything ? Looks like we're only doing a string comparison in there and some tracing and we made sure to check `process` is not null first.

Reply:
You are right. It's probably pointless now.